src/Storage/Device/S3.php

```diff
@@ -191,8 +301,11 @@ public function upload($source, $path, $chunk = 1, $chunks = 1): int
     public function read(string $path, int $offset = 0, int $length = null): string
     {
         $uri = ($path !== '') ? '/' . \str_replace('%2F', '/', \rawurlencode($path)) : '/';
+        if($length !== null) {
+            $end = $offset + $length - 1;
+            $this->headers['range'] = "bytes=$offset-$end";
```

Comment:
Is this the intended string value for the `range` header?
`['range' => 'bytes=$offset-$end']`

Reply:
Yes, it is. If reading a chunk the range will provide which part of the file was read.